src/index.js

```diff
@@ -1,21 +1,33 @@
 "use strict";
 
-let WritableStream = require("./writable_stream");
+const WritableStream = require("./writable_stream");
 
-module.exports = class Renderer {
-	constructor(bundlePath) {
-		this.bundlePath = bundlePath;
-		this.reload();
-	}
+module.exports = rendererPath => {
+	return (req, res, next) => {
+		const renderer = typeof rendererPath === "function" ?
+			rendererPath :
+			resolveRenderer(res.app, rendererPath, { cache: true });
 
-	render(response, tag, params) {
-		// TODO: `response.status(200);`?
-		let stream = new WritableStream(response);
-		this._render(stream, tag, params);
-		response.end();
-	}
+		res.complate = (tag, params) => {
+			let stream = new WritableStream(res);
+			renderer(stream, tag, params);
+			res.end();
+		};
 
-	reload() {
-		this._render = require(this.bundlePath);
-	}
+		next();
+	};
 };
+
+function resolveRenderer(app, rendererPath, { cache } = {}) {
+	// by default the require cache is disabled for dev environment,
+	// so the given renderer is updated on each request.
+	// This enables a watch/compile setup for re-creating the renderer
+	// and its components.
+	//
+	// Set `cache` to true, if you want to cache the renderer also in dev-mode.
+	if(!cache && !app.enabled("view cache")) {
+		delete require.cache[rendererPath];
+	}
+
+	return require(rendererPath);
+}
```

Comment:
as a side note. I would stay by 'rendererPath' instead of 'bundlePath' while that is the crucial part about that content, right? 'bundlePath' gives not really a hint about what the bundle is expected to contain (while that might be obvious)